config/routes.php

```diff
@@ -29,6 +31,24 @@
 	 * plugins.
 	 */
 	$before = [
+		// handle CORS preflight requests
+		[
+			'pattern' => '(:all)',
+			'method'  => 'OPTIONS',
+			'action'  => function () use ($kirby) {
+				if ($kirby->isCorsEnabled() === false) {
+					return null;
+				}
+
+				$headers = Cors::headers(preflight: true);
+
+				if ($headers === []) {
+					return null;
+				}
+
+				return new Response('', null, 204, $headers);
```

Comment:
`Question`: I wonder if we should rather use `HTTP 200` here. I stumbled across this on MDN, but not sure if it's (still) relevant.

Reference: https://developer.mozilla.org/en-US/docs/Web/HTTP/Reference/Methods/OPTIONS#:~:text=204%20No%20Content%20applies%20to%20the%20resource

Reply:
Good catch. From my understanding, `204 No Content` is more semantically correct for a CORS preflight because the server is only confirming permissions via headers and not returning a representation. Some old or nonstandard clients mishandle `204`, that's why Mozilla notes it.

The modern frameworks I know of use `204` across the board.